hail/src/main/scala/is/hail/backend/Backend.scala

```diff
@@ -172,39 +172,39 @@ abstract class Backend {
   def withExecuteContext[T](methodName: String)(f: ExecuteContext => T): T
 
   final def valueType(s: String): Array[Byte] =
-    withExecuteContext("valueType") { ctx =>
-      val v = IRParser.parse_value_ir(s, IRParserEnvironment(ctx, irMap = persistedIR.toMap))
-      v.typ.toString.getBytes(StandardCharsets.UTF_8)
+    jsonToBytes {
+      withExecuteContext("valueType") { ctx =>
+        IRParser.parse_value_ir(s, IRParserEnvironment(ctx, irMap = persistedIR.toMap)).typ.toJSON
+      }
     }
```

Comment:
Have you tested that this doesn't change the encoding? E.g. does it put quotes around the string?

Reply:
Good point - I'll change how it's deserialised in python. It is covered by existing tests.

Reply:
no. it's not. I can't see where there are used?

Reply:
ok added round-robin tests and found a few legit bugs. was miserable.

Reply:
Nice bug catches! Thanks for cleaning that up!